### PR TITLE
Getting Google Analytics Metrics 

### PR DESCRIPTION
**Pull request by cansavvy**
#### What changes are being implemented in this Pull Request?

Goals: We want any useful stats from Google Analytics about our many many websites and courses. 

#### What was your approach?

Ideally we can just call one function that collects an entire GitHub organization's repos' metrics. 

#### Usage 

```
authorize("google") # Login with an account that actually has a Google Analytics account -- this is key
accounts <- get_ga_user()

# Pick an account
account_id <- accounts$id[5]
stats_list <- all_ga_metrics(account_id = account_id)
```

### CODE REVIEW

File: R/google-analytics.R [hunk 1]
```diff
@@ -31,3 +31,143 @@ get_ga_user <- function() {
...
+#' @examples \dontrun{
+#'
+#' authorize("google")
+#' accounts <- get_ga_user()
+#'
+#' accounts$id[1]
+#' properties_list <- get_ga_properties(account_id = accounts$id[1])
+#'
+#' }
+get_ga_metrics <- function(property_id, start_date = "2015-08-14", end_date = lubridate::today()) {
```
Comment (cansavvy) [inline R/google-analytics.R:133]: This is not yet working. Not sure why. It's giving a 404 but on the Google API explorer the url looks the same. Need to investigate further.

File: R/google-analytics.R [hunk 2]
```diff
@@ -31,3 +31,143 @@ get_ga_user <- function() {
...
+
+  # Get auth token
+  token <- get_token(app_name = "google")
+  config <- httr::config(token = token)
+
+  body_params <- list(
+    "dateRanges" = list(
+      "startDate" = start_date,
+      "endDate" = end_date),
+    list(name = "activeUsers"),
```
Comment (cansavvy) [inline R/google-analytics.R:147]: This is a list of the stats I think are useful. I think it makes sense to functionalize this in such a way that allows folks to provide their own desired metrics they'd like to use. But that will probably be a future PR.

File: R/google-analytics.R (hunk 1 above)
Comment (cansavvy) [reply R/google-analytics.R]: Fixed!

**Comment by howardbaek** [comment]
What do you think of outputting tibbles instead of dataframes? The googlesheets4 package does this here https://github.com/tidyverse/googlesheets4#read , and it looks much nicer in the console.

**Comment by howardbaek** [comment]
The Usage functions work, but I think the trick now is to coerce those unpretty lists into readable tibbles.

**Comment by cansavvy** [comment]
Yes! That is what #16 and #17 are!

**Review by howardbaek — APPROVED** [review]

**Pull request merged**